### DISCUSSION TITLE
Different users, identical emails

### DISCUSSION THREAD

**Discussion by moshekatz** [supabase/supabase#1053] [Questions] [answered]
Hey,
I noticed that my users table has two different users with identical email.

The only difference is one Capital letter:
![image](https://user-images.githubusercontent.com/25473107/113503628-d1c6ae00-953b-11eb-9bd4-f9ad9c6a7b99.png)

I'm afraid to delete the duplicated one as one is an actual user with real data attached, and I don't want to accidently delete both.
I also don't want to ignore this.

Whould love to get a suggestion.
Thanks!

Regardless - you guys made an awesome (really awesome) job with this platform.
My go-to from now on.

**Comment by silentworks** [comment, marked as answer]
@moshekatz one way to make sure this doesn't happen in the future is to transform emails to lowercase always. I don't think email systems are case sensitive so the emails should always be lowercase characters.

For your current case, you can delete by id the one that is incorrect from your database. Or if you have a GUI editor like TablePlus or pgAdmin you can browser to the record and delete it manually.

**Reply by moshekatz** [reply]
Will do, thanks!

**Reply by charlie17** [reply]
Seems like it would be a good idea for Supabase to prevent such mixed-case / duplicate emails from getting into `auth.users`?

I added the following to prevent it in my app:

```
CREATE UNIQUE INDEX idx_lower_unique_email 
   ON auth.users (lower(email))
```

This will actually store whatever the user entered (upper or lower), but enforce case-insensitive uniqueness within the table.
